db/blip_sync_context.go

```diff
@@ -617,8 +620,26 @@ func (bsc *BlipSyncContext) sendNoRev(sender *blip.Sender, docID, revID string,
 // Pushes a revision body to the client
 func (bsc *BlipSyncContext) sendRevision(sender *blip.Sender, docID, revID string, seq SequenceID, knownRevs map[string]bool, maxHistory int, handleChangesResponseCollection *DatabaseCollectionWithUser, collectionIdx *int) error {
 	rev, err := handleChangesResponseCollection.GetRev(bsc.loggingCtx, docID, revID, true, nil)
+
+	var replacedRevID string
+
+	collectionCtx, collectionErr := bsc.collections.get(collectionIdx)
+	if collectionErr != nil {
+		return collectionErr
+	}
+
+	if collectionCtx.sendReplacementRevs && base.IsDocNotFoundError(err) {
+		base.DebugfCtx(bsc.loggingCtx, base.KeySync, "Unavailable revision for %q %s - finding replacement: %v", base.UD(docID), revID, err)
+
+		// try the active rev instead as a replacement
+		replacementRev, replacementRevErr := handleChangesResponseCollection.GetRev(bsc.loggingCtx, docID, "", true, nil)
+		// set replacement and continue as normal
+		replacedRevID = revID
+		revID = replacementRev.RevID
```

Comment:
If replacementRevErr is 'not found', replacementRev could be nil?  (or at least RevID won't be set)